File: QID-3525-SFSbottomstraddle/SFSbottomstraddle.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

straddle = function(St, K, r, T, sigma) {
    
    # Calculate the terms for the BS option prices
    d1 = (log(St/K) + (r + sigma^2/2) * T)/(sigma * sqrt(T))
    d2 = d1 - sigma * sqrt(T)
    
    # Calculate the plain vanilla option prices
    cal = St * pnorm(d1) - K * exp(-r * T) * pnorm(d2)
    put = -St * pnorm(-d1) + K * exp(-r * T) * pnorm(-d2)
    
    # Value of plain vanilla options at time T
    cal_T = cal * exp(r * T)
    put_T = put * exp(r * T)
    
    # Set the coordinates
    x = c(0, K, 2 * K)
    
    # Calculate the pay-off at each coordinate
    y1 = c(-cal_T, -cal_T, K - cal_T)
    y2 = c(K - put_T, -put_T, -put_T)
    
    # Determine the strategy pay-off
    y = y1 + y2
    
    # Plot straddle option strategy pay-off
    plot(x, y, type = "l", lwd = 3, col = "red", xlab = "S_T", ylab = "Payoff", xlim = c(0, 
        2 * K), ylim = c(-1.3 * (cal_T + put_T), 1.3 * (K - min(cal_T, put_T))))
    title("Bottom Straddle")
    
    # Plot plain vanilla option payoff profiles
    lines(x, y1, lty = 2)
    lines(x, y2, lty = 2)
    lines(x, c(0, 0, 0), lty = 3)
}

straddle(20, 25, 0.03, 1, 0.4) 

File: QID-3525-SFSbottomstraddle/SFSbottomstraddle.m
% ---------------------------------------------------------------------
% Book:        SFS
% ---------------------------------------------------------------------
% Quantlet:    SFSbottomstraddle
% ---------------------------------------------------------------------
% Description: SFSbottomstraddle plots the combination of a long call 
%              and a long put with the same strike, i.e. a straddle strategy. Values can be
%              entered interactively. Refers to exercise 1.4 in SFS.
% ---------------------------------------------------------------------
% Usage:       SFSbottomstraddle
% ---------------------------------------------------------------------
% Inputs:      St    - Stock price
%              K     - Strike price
%              r     - Interest rate
%              T     - Time to expiration
%              sigma - Volatility
% ---------------------------------------------------------------------
% Output:      Plot of a straddle option strategy
% ---------------------------------------------------------------------
% Example:     An example is produced for the values: St=20, K=25, T=1, 
%              sigma = 0.4 and r=0.03.
% ---------------------------------------------------------------------
% Author:      Lasse Groth 20090709
% ---------------------------------------------------------------------

clear
clc
close all

disp('Please input St, K, T, sigma, r  as: [20,25,1,0.4,0.03]') ;
disp(' ') ;
para = input('[St, K, T, sigma, r]=');

while length(para) < 5
    disp(' ')
    disp('Not enough input arguments. Please input in 1*5 vector form like [20,25,1,0.4,0.03]');
  para = input('[St, K, T, sigma, r]=');
end

St    = para(1);
K     = para(2);
T     = para(3);
sigma = para(4);
r     = para(5);
disp(' ');

[call, put] = blsprice(St,K,r,T,sigma);

call_T      = call*exp(r*T);
put_T       = put*exp(r*T);

x           = [0 K 2*K];

y1          = [-call_T -call_T K-call_T];
y2          = [K-put_T -put_T -put_T];
y3          = y1+y2;

hold on
plot(x,y1,'--k','LineWidth',1)
plot(x,y2,'--k','LineWidth',1)
plot(x,y3,'-r','LineWidth',2)
plot(x,[0 0 0],':k','LineWidth',0.2)
hold off

axis([0 2*K -1.3*(call_T+put_T) 1.3*(K-min(call_T,put_T))])
xlabel('S_T');      
ylabel('Payoff');
title('Bottom Straddle');

